mozphab/jujutsu.py

```diff
@@ -0,0 +1,433 @@
+        ).rstrip()
+        self.__email = check_output(
+            ["jj", "config", "get", "user.email"], split=False
+        ).rstrip()
+        self.__previous_change_ids = None
+        self.__change_id_at_start_was_empty = None
+
+    def check_vcs(self) -> bool:
+        if self.args.force_vcs:
+            return True
+
+        if self.is_cinnabar_required and not self.is_cinnabar_installed:
+            raise Error(
+                "Git Cinnabar extension is required to work on this repository."
+            )
+
+        return True
+
+    def before_submit(self):
+        self.__validate_email()
+
+        current_revision_is_empty = None
+        try:
+            current_revision_is_empty = self.__cli_log(
+                revset="@", template="empty", split=False
+            ).rstrip()
+        except:
+            pass
+        if current_revision_is_empty not in ["false", "true"]:
+            raise Error(
+                "Failed to parse current revision for unknown reasons."
+            )  # TODO: casing, etc. consistent?
+        self.__change_id_at_start_was_empty = current_revision_is_empty == "true"
+
+        try:
+            revset = "@-" if self.__change_id_at_start_was_empty else "@"
+            self.__previous_change_ids = self.__cli_log(
+                revset=revset, template="change_id"
+            )
+        except Exception:
+            raise Error("Jujutsu failed to read the branch name for unknown reasons.")
+
+    def set_args(self, args: argparse.Namespace):
+        """Store moz-phab command line args and set the revset."""
+        super().set_args(args)
+
+        # TODO: Did we accidentally change the behavior of `--single` with no `start_rev` specified?
+        is_single = hasattr(self.args, "single") and self.args.single
+
+        # TODO: What if we try to submit something that's immutable?
+        start_rev = None
+        if (
+            hasattr(self.args, "start_rev")
+            and self.args.start_rev != environment.DEFAULT_START_REV
+        ):
+            start_rev = self.args.start_rev
+        else:
+            start_rev = "@-" if is_single else self.__get_first_mutable_change()
+        if not start_rev:
+            return None
+
+        end_rev = None
+        if is_single:
+            end_rev = start_rev
+        elif (
+            hasattr(self.args, "end_rev")
+            and self.args.end_rev != environment.DEFAULT_END_REV
+        ):
+            end_rev = self.args.end_rev
+        else:
+            end_rev = start_rev
+
+        self.revset = (start_rev, end_rev)
+
+    def commit_stack(self, single: bool = False) -> Optional[List[Commit]]:
+        """Collect all the info about commits."""
+        logger.debug(f"searching with start and end at {self.revset}")
+        if not self.revset:
+            # No commits found to submit
+            return None
+
+        boundary = '"--%s--\n"' % uuid.uuid4().hex
```

Comment:
```suggestion
        boundary = "--%s--\n" % uuid.uuid4().hex
```

Reply:
The log won't get split properly later if the quotation marks are included directly in `boundary`

Reply:
Oh, shoot, good catch! 🙂

Reply:
Fixed with a force-push of the relevant commit to 58d7101d.